igvm/hypervisor.py

```diff
@@ -215,6 +190,40 @@ def check_vm(self, vm, offline):
                     .format(*hw_pair)
                 )
 
+        # HV in supported state?
+        if self.dataset_obj['state'] not in ['online', 'online_reserved']:
+            raise InvalidStateError(
+                'Hypervisor "{}" is not in online state ({}).'
+                .format(self.fqdn, self.dataset_obj['state'])
+            )
+
+        # Enough CPUs?
+        if vm.dataset_obj['num_cpu'] > self.dataset_obj['num_cpu']:
```

Comment:
I think we allow CPU over allocation no ?

Reply:
Noppe.

Reply:
lets wait for feedback from @TuxPowered42 but addresse this in another pr since the logic is the same as before, just in another place